src/components/master/ManageSport.jsx

```diff
@@ -0,0 +1,410 @@
+  const [selectedUserName, setSelectedUserName] = useState(null);
+  const { sportData } = useSelector(manageSportDataSelector);
+  console.log(sportData, "dataa");
+
+  const openModal = (id, type) => {
+    if (type === "delete") {
+      setModalContent("Do you want to delete this record?");
+      setAction(() => async () => {
+        try {
+          const response = await SportDeleteApi({ sportId: id }).unwrap();
+          if (response?.code === 200) {
+            dispatch(
+              handleNotification({
+                state: true,
+                message: response?.message,
+                severity: response?.code,
+              })
+            );
+          } else {
+            dispatch(
+              handleNotification({
+                state: true,
+                message: response?.message,
+                severity: response?.code,
+              })
+            );
+          }
+        } catch (error) {
+          dispatch(
+            handleNotification({
+              state: true,
+              message: sportDeleteData?.message,
+              severity: sportDeleteData?.code,
+            })
+          );
+        }
+      });
+    }
+    setModal(true);
+  };
+  const closeModal = () => {
+    setModal(false);
+  };
+
+  const [
+    addUpdateSport,
+    { data: addUpdateSportData, isSuccess: updataSportSuccess },
+  ] = useGetAddUpdateSportApiByNameMutation();
+
+  const [
+    userListSport,
+    {
+      data: listSportData,
+      isSuccess: userSportSuccess,
+      isLoading: sportDataFetching,
+    },
+  ] = useGetUserListSportApiByNameMutation();
+
+  const [
+    sendSportNotify,
+    { data: sendSportData, isSuccess: sendSportSuccess },
+  ] = useGetSendSportNotificaticationApiByNameMutation();
+
+  const [
+    userSetInvite,
+    { data: setInviteButtonData, isSuccess: setInviteSuccess },
+  ] = useGetSetInviteAndCompButtonApiByNameMutation();
+
+  console.log(setInviteButtonData, "ASD");
+  const [
+    SportDeleteApi,
+    {
+      data: sportDeleteData,
+      isLoading: sportDeleteLoading,
+      error: sportDeleteError,
+      isSuccess: sportDeleteSuccess,
+    },
+  ] = useDeleteSportByNameMutation();
+
+  const handleEditClick = (rowData) => {
+    console.log(sportData, "data");
+
+    const stackValue = sportData?.data?.filter((e) => {
+      if (e._id === rowData?.rowData[6]) {
+        return e;
+      }
+    });
+    const formattedStartDate = moment(rowData?.rowData[1]).format("L");
+
+    const formattedEndDate = moment(rowData?.rowData[2]).format("L");
+    const payload = [
+      {
+        sportname: rowData?.rowData[0],
+        startDate: formattedStartDate,
+        endDate: formattedEndDate,
+        type: rowData?.rowData[3],
+        description: rowData?.rowData[4],
+        bonus: rowData?.rowData[5],
+        id: rowData?.rowData[6],
+        stack: stackValue?.length ? stackValue[0].stack : "",
+      },
+    ];
```

Comment:
use mapping here create an array of these keys in constant file and do mapping here